script.js

```diff
@@ -1,4 +1,215 @@
+var gameMode = "Waiting for Number of Players";
+var gameStart = false;
+var playerScores = [];
+const winnerList = [];
+var noOfPlayers = 0;
+var currentPlayer = 0;
+var playerDiceRolls = [];
+var previousScores = []; //counter for round wins
+var previousWinners = []; // counter for winners each round
+var gamePlayMode = "Choose Play Mode";
+var noOfDice = 2;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  myOutputValue = "";
+  //choose number of players to play
+  if (!gameStart && gameMode == "Waiting for Number of Players") {
+    if (input <= 1 || isNaN(input)) {
+      myOutputValue = "At least 2 players are required for this game";
+    } else {
+      // if input is validated (players >=2), user to choose the game mode (highest or lowest dice)
+      gameMode = "Choose Game Mode";
+      noOfPlayers = input;
+      for (var i = 0; i < noOfPlayers; i++) {
+        previousWinners.push(0);
+      }
+      console.log(previousWinners);
+      myOutputValue = `🎲 Game started with ${input} players. Play in reverse mode? Y/N. 🎲`;
+    }
+  }
+  // select game mode (reverse = lowest dice or normal = highest dice)
+  else if (
+    !gameStart &&
+    gameMode == "Choose Game Mode" &&
+    gamePlayMode == "Choose Play Mode"
+  ) {
+    //player chooses Lowest Dice Mode
+
+    // can force input to all lowercase**
+
+    if (input == "Y" || input == "y") {
+      gameMode = "Choose Number of Dice";
+      gamePlayMode = "Lowest Dice";
+      return `Lowest 🎲 Game started with ${noOfPlayers} players. How many 🎲 would you like to play with?`;
+    } else if (input == "N" || input == "n") {
+      //player chooses Highest Dice Mode
+      gameMode = "Choose Number of Dice";
+      gamePlayMode = "Highest Dice";
+      return `Highest 🎲 Game started with ${noOfPlayers} players. How many 🎲 would you like to play with?`;
+    } else {
+      return `Please input Y to start the game in reverse mode, or N to play in normal mode.`;
+    }
+    //player chooses number of dice to use
+  } else if (!gameStart && gameMode == "Choose Number of Dice") {
+    if (input <= 1 || isNaN(input)) {
+      return `At least 2 dices are required for this game`;
+    } else {
+      noOfDice = input;
+      gameMode = "Roll Dice";
+      gameStart = true;
+      return `Game started with ${noOfDice} 🎲 . Click Submit to start rolling.`;
+    }
+  }
+  //dice-roll mode for all players
+  else if (gameStart && gameMode == "Roll Dice") {
+    console.log(gamePlayMode);
+    if (currentPlayer < noOfPlayers) {
+      playerDiceRolls = [];
+      currentPlayer += 1;
+      var diceResult = beatThatRollDice(playerDiceRolls, currentPlayer);
+      return diceResult;
```

Comment:
Nice use of a helper function to return an output value that's already the stringed myOutputValue.